wled00/FX_2Dfcn.cpp

```diff
@@ -173,47 +173,44 @@ void IRAM_ATTR_YN Segment::setPixelColorXY(int x, int y, uint32_t col)
   if (!isActive()) return; // not active
   if (x >= virtualWidth() || y >= virtualHeight() || x<0 || y<0) return;  // if pixel would fall out of virtual segment just exit
 
-  uint8_t _bri_t = currentBri();
+  uint8_t _bri_t = currentBri(); 
   if (_bri_t < 255) {
     col = color_fade(col, _bri_t);
   }
 
   if (reverse  ) x = virtualWidth()  - x - 1;
   if (reverse_y) y = virtualHeight() - y - 1;
   if (transpose) { std::swap(x,y); } // swap X & Y if segment transposed
-
   x *= groupLength(); // expand to physical pixels
   y *= groupLength(); // expand to physical pixels
-
   int W = width();
   int H = height();
-  if (x >= W || y >= H) return;  // if pixel would fall out of segment just exit
-
-  uint32_t tmpCol = col;
+ 
+  int yY = y;
   for (int j = 0; j < grouping; j++) {   // groupping vertically
+    if(yY >= H) continue;
```

Comment:
`break` would do just as well as `yY` is only ever increased.

Reply:
good point.